tests/epyccel/test_functionals.py

```diff
@@ -1,43 +1,99 @@
+        out2 = self._f2(*args)
+        assert equal(out1, out2 ).all()
 
-def test_functional_for_1d_var():
+#==============================================================================
+
+def test_functional_for_1d_range(language):
+    test = epyccel_test(functionals.functional_for_1d_range, lang=language)
+    test.compare_epyccel()
+
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = [
+            pytest.mark.python,
+            pytest.mark.skip(reason = "Too many expressions introduced for 'for xi in x'")
+        ])
+    )
+)
+def test_functional_for_1d_var(language):
     y = randint(99,size = 4)
-    compare_epyccel(functionals.functional_for_1d_var, y)
+    test = epyccel_test(functionals.functional_for_1d_var, lang=language)
+    test.compare_epyccel(y)
 
-def test_functional_for_2d_range():
-    compare_epyccel(functionals.functional_for_2d_range)
+def test_functional_for_2d_range(language):
+    test = epyccel_test(functionals.functional_for_2d_range, lang=language)
+    test.compare_epyccel()
 
-def test_functional_for_2d_var_range():
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = [
+            pytest.mark.python,
+            pytest.mark.skip(reason = "Too many expressions introduced for 'for xi in x'")
+        ])
+    )
+)
+def test_functional_for_2d_var_range(language):
     y = randint(99, size = 3)
-    compare_epyccel(functionals.functional_for_2d_var_range, y)
+    test = epyccel_test(functionals.functional_for_2d_var_range, lang=language)
+    test.compare_epyccel(y)
 
-def test_functional_for_2d_var_var():
+@pytest.mark.parametrize( 'language', (
+        pytest.param("fortran", marks = pytest.mark.fortran),
+        pytest.param("c", marks = pytest.mark.c),
+        pytest.param("python", marks = [
+            pytest.mark.python,
+            pytest.mark.skip(reason = "Too many expressions introduced for 'for xi in x'")
+        ])
+    )
+)
+def test_functional_for_2d_var_var(language):
     y = randint(99, size = 3)
     z = randint(99, size = 2)
-    compare_epyccel(functionals.functional_for_2d_var_var, y, z)
+    test = epyccel_test(functionals.functional_for_2d_var_var, lang=language)
+    test.compare_epyccel(y, z)
 
-def test_functional_for_2d_dependant_range():
-    compare_epyccel(functionals.functional_for_2d_dependant_range_1)
-    compare_epyccel(functionals.functional_for_2d_dependant_range_2)
-    compare_epyccel(functionals.functional_for_2d_dependant_range_3)
+def test_functional_for_2d_dependant_range(language):
+    test = epyccel_test(functionals.functional_for_2d_dependant_range_1, lang=language)
+    test.compare_epyccel()
+    test.compare_epyccel()
+    test.compare_epyccel()
```

Comment:
Why are we not testing the functions `functional_for_2d_dependant_range_2` and `functional_for_2d_dependant_range_3` anymore?

And what is the point of comparing the results multiple times if we do not change the input arguments?

Reply:
Ah sorry, I misread this and thought that it was the same test. That's why I added `epyccel_test` to this file. I have removed it. See d5f741d